### DISCUSSION TITLE
Disable WSLg permanently

### DISCUSSION THREAD

**Discussion by rlnt** [microsoft/wslg#523] [Q&A] [answered]
Hello! Good job on WSLg, it looks promising.
In my opinion, it still needs a bit of polishing before it's actually usable for me personally so I would like to get rid of it until I decide to use it again.

I uninstalled the `WSLg Preview` from my Windows apps but it keeps installing itself over and over.
Is there a way to completely disable it for now? I still want to keep my normal WSL2.

Before you say "Why don't you leave it on? It doesn't do anything if you don't use it.", I have to say that I want to use another way of displaying GUI apps for now which works better for my use case and WSLg is taking priority over it. So I need to disable it.

**Comment by jeremy-smith-maco** [comment, marked as answer]
Is this what you were after?

> User can also choose to turn off the system distro entirely by adding the following entry in their .wslconfig file (located at c:\users\MyUser\.wslconfig). This will turn off support for GUI applications in WSL.

```
[wsl2]
guiApplications=false
```

Reference: https://github.com/microsoft/wslg#wslg-system-distro

**Reply by rlnt** [reply]
I tried that and rechecked it, now that you mention it, but it seems it's just ignored and WSLg is reinstalled every reboot.

**Reply by jeremy-smith-maco** [reply]
I think WSLg is packaged with WSL by default now so I am not sure if you can avoid it being installed. I think all you can do is use that above config to stop your applications from using it unless there is some way to build WSL without WSLg in it.

**Reply by rlnt** [reply]
Looks like it was just a permission issue. I recreated the config and now it's not reinstalling automatically. Thank you, will mark it as answer.

**Reply by ghost** [reply]
> Looks like it was just a permission issue. I recreated the config and now it's not reinstalling automatically. Thank you, will mark it as answer.

what was the permission problem, I have a problem making it work

**Reply by rlnt** [reply]
The file was `read-only` somehow. After removing and recreating the file with the needed config entry, it worked fine. Had to restart my system tho if I remember correctly.

**Reply by ghost** [reply]
@DAmNRelentless unfortunately I'm still unable to remove it. if you can give me a hand, I'm really appreciate, like you, I'm the kind of guy that doesn't like having something on my computer that doesn't do anything for me
what do you think is the reason

you said it's a permission problem, these our my permissions results
```
> ICACLS .wslconfig
.wslconfig NT AUTHORITY\SYSTEM:(I)(F)
           BUILTIN\Administrators:(I)(F)
           VIVOBOOK\sinamobasheri:(I)(F)

Successfully processed 1 files; Failed processing 0 files
```
or in PowerShell
```
PS C:\Users\Sina> (Get-Acl .\.wslconfig).access | ft

FileSystemRights AccessControlType IdentityReference      IsInherited InheritanceFlags PropagationFlags
---------------- ----------------- -----------------      ----------- ---------------- ----------------
     FullControl             Allow NT AUTHORITY\SYSTEM           True             None             None
     FullControl             Allow BUILTIN\Administrators        True             None             None
     FullControl             Allow VIVOBOOK\sinamobasheri        True             None             None
```
.wslconfig encoding is UTF-8 with CRLF end of the line, is there something wired about this file? @jeremy-smith-maco

**Reply by rlnt** [reply]
You can't uninstall it completely. That would probably require some registry hacks.
It will automatically reinstall itself, the only thing you can do is disable it through the config so its commands won't be registered anymore. That's also the state I currently have.

**Reply by Masamune3210** [reply]
If you dont like having things on your computer that don't do anything for you specifically, then you arent going to want to look at the rest of windows lol

**Comment by Iano-theDev** [comment]
is this file the same as this one located: /etc/wsl.conf  ?

**Reply by rlnt** [reply]
No, the `/etc/wsl.conf` holds settings like whether the system should boot with systemd.
The file where this setting belongs is located in your user profile folder on your Windows system.

You can access the directory by pressing `Win` + `R` and then entering `%userprofile%`. The folder that will open is the correct one. If you don't have a `.wslconfig` file in there yet, simply create one.

**Reply by Iano-theDev** [reply]
Alright, thanks for clarfying....
